learning_phase/chat_sample/server.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"context"
+	"log"
+	"net"
+	"os"
+	"os/signal"
+	"strings"
+	"sync"
+)
+
+type server struct {
+	Password          string
+	ClientConnections map[string]*client
+	commands          chan command
+	mu                sync.RWMutex
+}
+
+func newServer(password string) *server {
+	return &server{
+		Password:          password,
+		ClientConnections: make(map[string]*client),
+		commands:          make(chan command),
+	}
+}
+
+func (s *server) run() {
+	// keeps receiving the commands from all clients
+	for cmd := range s.commands {
+		switch cmd.id {
+		case cmdauth:
+			s.auth(cmd.client, cmd.args[1], cmd.args[2])
+		case cmdpm:
+			s.pm(cmd.client, cmd.args)
+		case cmdmsg:
+			s.msg(cmd.client, cmd.args)
+		case cmdquit:
+			s.quit(cmd.client)
```

Comment:
I think the loop may still continue after quit() is called

Reply:
okay so here the quit is for the client to cancel connection, we delete the clients details from the map in that function. So shouldn't the server keep running since there are other clients left.